app/gui/src/project-view/components/visualizations/TableVisualization.vue

```diff
@@ -625,6 +625,26 @@ const getColumnValueToEnso = (columnName: string) => {
   if (isNumber.indexOf(columnType) != -1) {
     return (item: string, module: Ast.MutableModule) => Ast.tryNumberToEnso(Number(item), module)!
   }
+  const createDateTimePattern = (pattern: string, numberOfParts: number) => {
+    const dateOrTimePattern = Pattern.parseExpression(pattern)
+    return (item: string, module: Ast.MutableModule) => {
+      const dateTimeParts = item.match(/\d+/g)!.map(Number)
+      const dateTimePartsNumeric = []
+      for (let i = 0; i < numberOfParts; i++) {
+        dateTimePartsNumeric.push(Ast.tryNumberToEnso(Number(dateTimeParts[i] ?? 0), module)!)
+      }
+      return dateOrTimePattern.instantiateCopied(dateTimePartsNumeric)
+    }
+  }
+  if (columnType === 'Date') {
+    return createDateTimePattern('(Date.new __ __ __)', 3)
+  }
+  if (columnType === 'Date_Time') {
+    return createDateTimePattern('(Date_Time.new __ __ __ __ __ __ __ __ __)', 9)
+  }
+  if (columnType === 'Time') {
+    return createDateTimePattern('(Time_Of_Day.new __ __ __ __ __ __)', 6)
+  }
```

Comment:
Can we rely on each of these types always being formatted the same way? What about timezones?

Reply:
In terms of formatting no matter what the input string '1/2/1997', '01 02 97' etc it seems to be that when displayed as a date column ag grid formats them all the same way (yyyy-mm-dd ...).
But I didn't check timezones so I'll test that scenario now